shell/platform/darwin/macos/framework/Source/FlutterPlatformViews.h

```diff
@@ -0,0 +1,62 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FLUTTERPLATFORMVIEWS_H_
+#define FLUTTER_FLUTTERPLATFORMVIEWS_H_
+
+#import <AppKit/AppKit.h>
+
+#import "FlutterCodecs.h"
+#import "FlutterMacros.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * Wraps a `NSView` for embedding in the Flutter hierarchy
+ */
+@protocol FlutterPlatformView <NSObject>
+/**
+ * Returns a reference to the `NSView` that is wrapped by this `FlutterPlatformView`.
+ *
+ * It is recommended to return a cached view instance in this method.
+ * Constructing and returning a new NSView instance in this method might cause undefined behavior.
```

Comment:
"Undefined behavior"? That's extremely bad if true in the technical sense of the term. What's the reason for that?

Also, this comment makes it sound like making and caching it on demand in this method isn't allowed, which I assume isn't actually the case?